__test__/bread-n-butter.test.ts

```diff
@@ -179,7 +179,11 @@ test("skip", () => {
 });
 
 test("choice", () => {
-  const abc = bnb.choice(bnb.text("a"), bnb.text("b"), bnb.text("c"));
+  const abc: bnb.Parser<"a" | "b" | "c"> = bnb.choice(
+    bnb.text("a"),
+    bnb.text("b"),
+    bnb.text("c")
+  );
```

Comment:
I think it would be good if `choice` tested the type parameter change you made:


```ts
  const abc123 = bnb.choice(
    bnb.text("a"),
    bnb.text("b"),
    bnb.text("c"),
    bnb.text("1").map(() => 1 as const),
    bnb.text("2").map(() => 2 as const),
    bnb.text("3").map(() => 3 as const)
  );
  snapTest(abc123, "a");
  snapTest(abc123, "b");
  snapTest(abc123, "c");
  snapTest(abc123, "1");
  snapTest(abc123, "2");
  snapTest(abc123, "3");
  snapTest(abc123, "aaaa");
  snapTest(abc123, "abb");
  snapTest(abc123, "");
```

This way you can see the type being detected correctly:

```ts
const abc123: bnb.Parser<2 | "a" | 1 | "b" | "c" | 3>
```